### DISCUSSION TITLE
Extracting Inclination using EBAI

### DISCUSSION THREAD

**Discussion by AjlaTr** [phoebe-project/phoebe2#880] [Questions]
Good day,

My team and I are attempting to extract inclination from normalized light curve data for an SB1 binary system. We have tried using the EBAI estimator's mlp model to compute the inclination angle but continuously run into the issue where the exact same inclination angle is given amongst vastly different star systems. We were hoping to have input provided on this issue.

Thank you for your time.

The results we got are attached as well.
![Screenshot 2024-06-02 120840](https://github.com/phoebe-project/phoebe2/assets/167161028/5bb4e496-4033-4f61-8495-1458cbfb5489)

      Code:
      `import phoebe
      from phoebe import u # units
      import numpy as np
      import matplotlib.pyplot as plt
      
      logger = phoebe.logger()
      phoebe.logger('error')
      b = phoebe.default_binary()
      
      import pandas as pd
      import seaborn as sns
      sns.set_theme()

      lcdata = pd.read_csv('lightcurve.csv')
      
      times = np.array(lcdata["hjd"])
      fluxes = np.array(lcdata["flux"])
      error = np.array(lcdata["flux_err"])
      
      meanlc = np.mean(fluxes)
      stdlc = np.std(fluxes)
      fluxes = (fluxes - meanlc)/stdlc
      min_flux = np.abs(fluxes.min())
      fluxes += min_flux
      mean_err_lc = np.mean(error)
      std_err_lc = np.std(error)
      error = np.abs((error-mean_err_lc)/std_err_lc)
      
      b["ecc"] =0.03333
      b["per0"] = 2.65739
      b["period@binary@component"] = 562.50768
      
      b.add_dataset('lc', times=times, fluxes=fluxes, sigmas = error)
      
      b["passband@lc01@dataset"] = "Tycho:B"
      
      b.set_value_all('ld_mode', 'manual')
      b.set_value_all('ld_mode_bol', 'manual')
      b.set_value_all('atm', 'blackbody')
      b.set_value('pblum_mode', 'dataset-scaled')
      
      try:
          b.run_compute(model= "first_try")
      except Exception as err:
          print(err)
      
      axs, artists = b["lc01"].plot( x= "phases",show = True, legend = True,  draw_sidebars=True)
      
      b.add_solver('estimator.ebai', ebai_method='mlp', solver='ebai_mlp')
      print(b.filter(solver='ebai_mlp'))
      
      b['phase_bin@ebai_mlp'] = False
      print(b.filter(solver='ebai_mlp'))
      
      b.run_solver('ebai_mlp', solution='ebai_mlp_solution')
      
      try:
          b.adopt_solution('ebai_mlp_solution')
      except Exception as e:
          print(e)
      
      b.flip_constraint('requivsumfrac', solve_for='requiv@secondary')
      
      b.flip_constraint('teffratio', solve_for='teff@secondary')
      b.flip_constraint('esinw', solve_for='ecc')
      b.flip_constraint('ecosw', solve_for='per0')
      
      b["ebai_mlp@solution@adopt_parameters"].value
      
      b["ebai_mlp@solution"]
      
      b.set_value('teff', component='primary', value=5004.9854)
      
      b['eclipse_method'] = 'native'
      
      print(b.adopt_solution('ebai_mlp_solution'))
      
      b.run_compute()
      
      b["lc01@latest"].plot(x='phase', ls='-', legend=True, show=True)
      
      b["lc01"].plot(x='phase', ls='-', legend=True, show=True)`

**Comment by aprsa** [comment]
EBAI is not a robust estimator because it doesn't provide an exhaustive coverage of the parameter space. This is especially evident in the case of SB1s. So the short answer is that EBAI is not suited for your use case; the reason why you're getting the same value out is neural network's way of saying "gee, I've no idea what this reminds me of, so I'll just give you some mean value of the sample set in the hope that I don't screw up too badly."

We have been working on the next generation of EBAI but it will be flipped: instead of estimating the parameters, it will allow you to sample the parameter space. This should be much better suited for your use case, but its precise ETA is still to be determined.

**Reply by AjlaTr** [reply]
Oh, I see. Thank you for your quick response. Do you believe that the phoebe emcee sampler is a feasible method for inclination?
